scripts-available/CDB_DistType.sql

```diff
@@ -0,0 +1,138 @@
+--
+-- CDB_DistType classifies the histograms of a column into
+-- one of the basic types listed by Galtong: http://druedin.com/2012/12/08/galtungs-ajus-system/
+-- 
+-- Future improvements:
+--    variable number of bins (7 is baked in right now)
+--    catch the number of items to ensure that the sample is large enough
+--
+-- Refs:
+--    1. width_bucket/histograms: http://tapoueh.org/blog/2014/02/21-PostgreSQL-histogram
+--    2. R implementation: https://github.com/cran/agrmt
+
+CREATE OR REPLACE FUNCTION CDB_DistType ( in_array NUMERIC[] ) RETURNS text as $$
```

Comment:
In exchange for another problem, which is "what if there are no stats for the column currently and the process running the function doesn't have privs to run ANALYZE"? Which is a deeper, system-level problem for CartoDB. If we're going to start leaning on the stats table for user-facing functionality, we need to take control over ensuring it is kept up-to-date. (Also, I need to check that stats are gathered for non-indexed columns, not sure that they are.)